crates/utils/re_video/src/stable_index_deque.rs

```diff
@@ -0,0 +1,322 @@
+use std::collections::VecDeque;
+
+/// A deque with an offset that allows indices into it to stay valid.
+///
+/// Does not expose the underlying [`VecDeque`] directly to guarantee that all operations are valid.
+/// This is useful to use stable indices for addressing in a growing/shrinking deque
+/// without having to resort to a more complex map datastructure.
+///
+/// To illustrate:
+/// ```
+/// // Indices for deque are unstable:
+/// # use std::collections::VecDeque;
+/// let mut v = (0..2).collect::<VecDeque<i32>>();
+/// assert_eq!(v.get(1), Some(&1));
+/// v.pop_front();
+/// assert_eq!(v.get(1), None);
+///
+/// // Indices for `StableIndexDeque` are stable:
+/// # use re_video::StableIndexDeque;
+/// let mut v = (0..2).collect::<StableIndexDeque<i32>>();
+/// assert_eq!(v.get(1), Some(&1));
+/// v.pop_front();
+/// assert_eq!(v.get(1), Some(&1));
+/// ```
+#[derive(Default, Clone, Debug)]
+pub struct StableIndexDeque<T> {
```

Comment:
Maybe `RangeDeque` would be a better name?

Reply:
I'm not happy with the current name either. But not gonna apply this for above reason: I think that's less descriptive since we're not dealing with ranges anywhere. RangeDeque sounds like a datastructure that allows pushing and popping in ranges and maybe has special handling for gaps in there.